src/datasets/dataset_dict.py

```diff
@@ -1417,6 +1423,61 @@ def push_to_hub(
             revision=branch,
         )
 
+    def to_pandas(
+        self,
+        splits: Optional[Union[Literal["all"], List[str]]] = None,
+        batch_size: Optional[int] = None,
+        batched: bool = False,
+    ) -> Union[pd.DataFrame, Iterator[pd.DataFrame]]:
+        """Returns the dataset as a :class:`pandas.DataFrame`. Can also return a generator for large datasets.
+
+        You must specify which splits to convert if the dataset is made of multiple splits.
+
+        Args:
+            splits (:obj:`Union[Literal["all"], List[str]]`, optional): List of splits to convert to a DataFrame.
+                You don't need to specify the splits if there's only one.
+                Use splits="all" to convert all the splits (they will be converted in the order of the dictionary).
+            batched (:obj:`bool`): Set to :obj:`True` to return a generator that yields the dataset as batches
+                of ``batch_size`` rows. Defaults to :obj:`False` (returns the whole datasets once)
+            batch_size (:obj:`int`, optional): The size (number of rows) of the batches if ``batched`` is `True`.
+                Defaults to :obj:`datasets.config.DEFAULT_MAX_BATCH_SIZE`.
+
+        Returns:
+            `pandas.DataFrame` or `Iterator[pandas.DataFrame]`
+
+        Example:
+
+        If the dataset has one split:
+        ```py
+        >>> df = dataset_dict.to_pandas()
+        ```
+
+        If the dataset has multiple splits:
+        ```py
+        >>> df_train = dataset_dict["train"].to_pandas()
+        >>> df_all = dataset_dict.to_pandas(splits="all")
+        >>> df_train_test = dataset_dict.to_pandas(splits=["train", "test"])
+        ```
+        """
+        self._check_values_type()
+        self._check_values_features()
+        if splits is None and len(self) > 1:
+            raise SplitsError(
```

Comment:
maybe invent a more specific name for this type of error? smth like `SplitsNotSpecifiedError`/`SplitsNotProvidedError` ? (subclassing `SplitsError`?)